packages/worker/src/worker.ts

```diff
@@ -1106,8 +1106,17 @@ export class Worker<T extends WorkerSpec = DefaultWorkerSpec> {
               this.state = 'STOPPED';
             },
             error: (error) => {
-              this.log.error('Worker failed', { error });
-              this.state = 'FAILED';
+              if (
+                (this.state == 'STOPPING' || this.state == 'DRAINING') &&
+                error instanceof errors.NoWorkerRegisteredError
+              ) {
+                // It's possible for the worker to be completely removed if we poll after completing
+                // everything, in which case it will be not registered.
+                this.state = 'STOPPED';
```

Comment:
Won't that not matter since it's already the last value in the combined observable of all the pollers/stuff?

Reply:
Basically both pollers either already have or are immediately about to encounter the not registered error.

In any case, I agree this would be better handled in core but I didn't want to have to do two merge rounds to wrap up this PR. I will do that in core and then clean this up.